File: Java/src/main/java/engisubs/gameobject/product/farm_product/FarmProduct.java
package main.java.engisubs.gameobject.product.farm_product;

import main.java.engisubs.gameobject.product.Product;

public class FarmProduct extends Product{
    protected String producer;

    public FarmProduct(){}

    public void initializeFarmProduct(String _producer){
        this.producer = _producer;
    }

    public String getProducer(){
        return this.producer;
    }

    public static FarmProductType getFarmProductType(GameObjectType objectType){
        if (objectType == GameObjectType.CHICKEN_EGG || objectType == GameObjectType.PLATYPUS_EGG || objectType == GameObjectType.PTERODACTYL_EGG) {
            return FarmProductType.EGG;
        } else if (objectType == GameObjectType.BISON_MEAT || objectType == GameObjectType.CHICKEN_MEAT || objectType == GameObjectType.DOG_MEAT || objectType == GameObjectType.PLATYPUS_MEAT || objectType == GameObjectType.TREX_MEAT){
            return FarmProductType.MEAT;
        } else if (objectType == GameObjectType.BISON_MILK || objectType == GameObjectType.PLATYPUS_MILK) {
            return FarmProductType.MILK;
        } else {
            return FarmProductType.NOT_FOUND;
        }
    }
}

File: CPP/src/FarmProduct.cpp
#include "../include/FarmProduct.hpp"

//Constructor 
FarmProduct::FarmProduct() {
    
}
void FarmProduct::initializeFarmProduct(string _producer) {
	producer = _producer;
}

FarmProductType FarmProduct::getFarmProductType(ObjectType objectType) {
	if (objectType == CHICKEN_EGG || objectType == PLATYPUS_EGG || objectType == PTERODACTYL_EGG) {
		return EGG;
	} else if (objectType == BISON_MEAT || objectType == CHICKEN_MEAT || objectType == DOG_MEAT || objectType == PLATYPUS_MEAT || objectType == TREX_MEAT){
		return MEAT;
	} else if (objectType == BISON_MILK || objectType == PLATYPUS_MILK) {
		return MILK;
	} else {
		return NOT_FOUND;
	}
}